cmd/exec-server/tls_config_test.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"net"
+	"os"
+	"testing"
+)
+
+const (
+	certPath   = "../../certs/"
+	clientKey  = certPath + "client-key.pem"
+	clientCert = certPath + "client-cert.pem"
+	serverKey  = certPath + "server-key.pem"
+	serverCert = certPath + "server-cert.pem"
+	testKey    = certPath + "test-key.pem"
+	testCert   = certPath + "test-cert.pem"
```

Comment:
These tests won't work if `make certs` hasn't been called. It's a good shortcut, but leave a note on how you would actually test this.

Reply:
My (main) makefile will ensure certs are created before running the tests too